[new product] Sourcegraph

Alexm0812 opened the pull request.

Alexm0812 committed:
add product sourcegraph.md

Alexm0812 committed:
add product sourcegraph.md

Alexm0812 committed:
Merge pull request #1 from Alexm0812/add-sourcegraph

add product sourcegraph.md

captn3m0 commented on products/sourcegraph.md:
```diff
@@ -0,0 +1,42 @@
+  methods:
+  - git: https://github.com/sourcegraph/sourcegraph.git
+    regex: ^v(?P<major>\d+)\.(?P<minor>\d+)\.(?P<patch>\d+)$
+    template: '{{major}}.{{minor}}.{{patch}}'
```
The default regex should work well enough.
```suggestion
```

captn3m0 commented on products/sourcegraph.md:
```diff
@@ -0,0 +1,42 @@
+eolColumn: Support
+
+identifiers:
+- purl: pkg:github/sourcegraph/sourcegraph
```
```suggestion
- purl: pkg:github/sourcegraph/sourcegraph
- purl: pkg:docker/sourcegraph/sg
```

captn3m0 commented on products/sourcegraph.md:
```diff
@@ -0,0 +1,42 @@
+
+- releaseCycle: "4"
+  releaseDate: 2022-09-22
+  eol: true
```
Based on releaseDate for 5.0

```suggestion
  eol: 2023-05-22
```

captn3m0 commented on products/sourcegraph.md:
```diff
@@ -0,0 +1,42 @@
+  eol: false
+  latest: "v5.3.3"
+  latestReleaseDate: 2024-03-20
+  link: https://github.com/sourcegraph/sourcegraph/releases
```
```suggestion
  link: https://github.com/sourcegraph/sourcegraph/releases/tag/v__LATEST__
```

captn3m0 commented on products/sourcegraph.md:
```diff
@@ -0,0 +1,42 @@
+- releaseCycle: "5"
+  releaseDate: 2023-05-22
+  eol: false
+  latest: "v5.3.3"
```
We ignore all version prefixes, as noted in CONTRIBUTING.md
```suggestion
  latest: "5.3.3"
```

captn3m0 commented on products/sourcegraph.md:
```diff
@@ -0,0 +1,42 @@
+  eol: true
+  latest: "4.5.1"
+  latestReleaseDate: 2023-02-24
+  link: https://github.com/sourcegraph/sourcegraph/releases
```
The release blog seems to be a better link:
```suggestion
  link: https://sourcegraph.com/blog/release/4.5
```

captn3m0 commented on products/sourcegraph.md:
```diff
@@ -0,0 +1,42 @@
+> [Sourcegraph](https://sourcegraph.com/) is a code search and intelligence tool for developers that helps
+> them find, understand, and fix code across all of their projects and dependencies.
+
+Only the latest version of each major release is supported.
```
While this does seem accurate (no 4.x release since 5.0 was published), we should either have a clear citation (should also be added as `releasePolicyLink`), or clearly mention that this is an estimate.

Secondly, clearly mention we are tracking self-hosted releases.

```suggestion
Sourcegraph is available in self-hosted and cloud editions. This page only tracks releases for
the self-hosted offering.

Looking at recent releases, it seems that only the latest version of each major release is supported.
```

captn3m0 commented on products/sourcegraph.md:
```diff
@@ -0,0 +1,42 @@
+tags: java-runtime
+permalink: /sourcegraph
+alternate_urls:
+- /sourcegraph-code-search
```
Not sure if this is helpful. We typically provide alternate_urls for urls that users might type and expect to work (k8s, eks etc). `/sourcegraph` is already a great URL, so don't see a need here. We could maybe add `/sg`, which seems to be used in a few places, but that is too ambiguous.

```suggestion
```

captn3m0 reviewed (changes requested):
thanks for the PR. left a few suggestions, happy to discuss further. The main one is the lack of an upstream release policy link.

Alexm0812 committed:
Update products/sourcegraph.md

Co-authored-by: Nemo <me@captnemo.in>

Alexm0812 committed:
Update products/sourcegraph.md

Co-authored-by: Nemo <me@captnemo.in>

Alexm0812 committed:
Update products/sourcegraph.md

Co-authored-by: Nemo <me@captnemo.in>

Alexm0812 committed:
Update products/sourcegraph.md

Co-authored-by: Nemo <me@captnemo.in>

Alexm0812 committed:
Update products/sourcegraph.md

Co-authored-by: Nemo <me@captnemo.in>

Alexm0812 committed:
Update products/sourcegraph.md

Co-authored-by: Nemo <me@captnemo.in>

Alexm0812 committed:
Update products/sourcegraph.md

Co-authored-by: Nemo <me@captnemo.in>

Alexm0812 committed:
Update products/sourcegraph.md

Co-authored-by: Nemo <me@captnemo.in>

Alexm0812 committed:
Update products/sourcegraph.md

Alexm0812 committed:
Merge pull request #2 from Alexm0812/add-sourcegraph

Update products/sourcegraph.md

Alexm0812 committed:
Update products/sourcegraph.md

Alexm0812 committed:
Merge pull request #3 from Alexm0812/add-sourcegraph

Update products/sourcegraph.md

Alexm0812 reviewed (commented):
All changes made

marcwrobel commented on products/sourcegraph.md:
```diff
@@ -0,0 +1,43 @@
+
+releases:
+- releaseCycle: "5"
+  releaseDate: 2023-05-22
```
```suggestion
  releaseDate: 2023-03-22
```

See https://github.com/sourcegraph/sourcegraph/releases/tag/v5.0.0.

marcwrobel commented on products/sourcegraph.md:
```diff
@@ -0,0 +1,43 @@
+category: server-app
+tags: java-runtime
+permalink: /sourcegraph
+releasePolicyLink: https://handbook.sourcegraph.com/departments/engineering/dev/process/releases/
```
```suggestion
releasePolicyLink: https://handbook.sourcegraph.com/departments/engineering/dev/process/releases/
changelogTemplate: https://github.com/sourcegraph/sourcegraph/releases/tag/v__LATEST__
```

Generated links work for both 4.x and 5.x.

marcwrobel commented on products/sourcegraph.md:
```diff
@@ -0,0 +1,43 @@
+  eol: false
+  latest: "5.3.3"
+  latestReleaseDate: 2024-03-20
+  link: https://github.com/sourcegraph/sourcegraph/releases/tag/v__LATEST__
```
```suggestion
```

We could use this as the main `changelogTemplate` as it works for both 4.x and 5.x. See my previous comment.

marcwrobel commented on products/sourcegraph.md:
```diff
@@ -0,0 +1,43 @@
+  eol: 2023-05-22
+  latest: "4.5.1"
+  latestReleaseDate: 2023-02-24
+  link: https://sourcegraph.com/blog/release/4.5
```
```suggestion
```

The main changelogTemplate works for both releases (https://github.com/sourcegraph/sourcegraph/releases/tag/v4.5.1). See my previous comment.

marcwrobel commented on products/sourcegraph.md:
```diff
@@ -0,0 +1,43 @@
+  methods:
+  - git: https://github.com/sourcegraph/sourcegraph.git
+
+releases:
```
```suggestion
# eol(x) = releaseDate(x+1)
releases:
```

For future maintainers.

marcwrobel commented on products/sourcegraph.md:
```diff
@@ -0,0 +1,43 @@
+Sourcegraph is available in self-hosted and cloud editions. This page only tracks releases for
+the self-hosted offering.
+
+Looking at recent releases, it seems that only the latest version of each major release is supported.
```
```suggestion
Looking at recent releases, it seems that only the latest release is supported.
```

Because you set the EOL date for 4.x to the release date of 5.x.

marcwrobel commented on products/sourcegraph.md:
```diff
@@ -0,0 +1,43 @@
+
+Looking at recent releases, it seems that only the latest version of each major release is supported.
+
+Sourcegraph provides [enterprise support](https://about.sourcegraph.com/contact/sales) with additional features for organizations.
```
```suggestion
Sourcegraph provides [enterprise support](https://sourcegraph.com/pricing) with additional features and support for organizations.
```

Because https://sourcegraph.com/pricing gives more explanation than just the contact page.

marcwrobel reviewed (changes requested):
Hi @Alexm0812, it's almost good to me, but I have a few more comments / suggestions.

Alexm0812 committed:
Update products/sourcegraph.md

Co-authored-by: Marc Wrobel <marc.wrobel@gmail.com>

Alexm0812 committed:
Update products/sourcegraph.md

Co-authored-by: Marc Wrobel <marc.wrobel@gmail.com>

Alexm0812 committed:
Update products/sourcegraph.md

Co-authored-by: Marc Wrobel <marc.wrobel@gmail.com>

Alexm0812 committed:
Update products/sourcegraph.md

Co-authored-by: Marc Wrobel <marc.wrobel@gmail.com>

Alexm0812 committed:
Update products/sourcegraph.md

Co-authored-by: Marc Wrobel <marc.wrobel@gmail.com>

Alexm0812 committed:
Update products/sourcegraph.md

Co-authored-by: Marc Wrobel <marc.wrobel@gmail.com>

Alexm0812 committed:
Update products/sourcegraph.md

Co-authored-by: Marc Wrobel <marc.wrobel@gmail.com>

Alexm0812 reviewed (commented):
Hi @marcwrobel thank you for the suggestions. I have committed these now and everything should be good.

captn3m0 committed:
[sourcegraph] Add note about upstream support

Since we are marking v4 as unsupported deliberately,
against what the upstream says - we need to be
clear why there is a disparity.

captn3m0 reviewed (approved):
LGTM. We can update further if we get clarity from upstream.

marcwrobel commented on products/sourcegraph.md:
```diff
@@ -0,0 +1,42 @@
+
+Sourcegraph is available in self-hosted and cloud editions. This page only tracks releases for
+the self-hosted offering. While limited support is offered for the previous major version, it does
+not include bug-fixes, or security backports. Consequentially , this page marks the older release as unsupported.
```
```suggestion
not include bug-fixes or security backports. Consequentially, this page marks all but the latest
release as unsupported.
```

captn3m0 replied:
I tried a bunch of variations, but nothing as good as this one. Thanks 💯

marcwrobel reviewed (approved):
:+1:

marcwrobel committed:
Update products/sourcegraph.md
